src/main/webapp/javascript/user-page.js

```diff
@@ -30,10 +29,42 @@ async function loadUserRecipes() {
           elementsToAddToImageDiv.forEach(elem => imageDivs[i].appendChild(elem));
         }
       }
+      addDeleteFunctionality(recipes);
+      // TODO: add edit functionality
     }
   }
 }
 
+function addDeleteFunctionality(recipes){
+  // functionality of delete button
+  const deleteButtons = document.getElementsByClassName('fa-trash-alt');
+  const recipeCards = document.getElementsByClassName('recipe-card');
+  // there are as many delete buttons as there are recipe cards
+  for (let i = 0; i < deleteButtons.length; i++) {
+    let recipe = recipes[i];
+    let recipeCard = recipeCards[i];
+    deleteButtons[i].addEventListener('click', () => {
+      const deleteConfirmed = confirm("Are you sure you want to delete the " 
+        + recipe.name + " recipe?\nThis action cannot be undone!");
+      if (deleteConfirmed) {
+        deleteRecipe(recipe);
+        // Remove the recipe from the DOM.
+        recipeCard.remove();
+      }
+      if (recipeCards.length == 0) {
+        noRecipes("user-recipes", "You have not uploaded any recipes yet.");
+      }
+    });
+  }
+}
+
+// takes in div id and message
+noRecipes = (divID, message) => {
```

Comment:
I know what this does, but I think that having an expanded comment for this function could improve readability in the future regarding use.